app/docker/services/imageService.js

```diff
@@ -138,6 +139,32 @@ angular.module('portainer.docker')
     return Image.tag({id: id, tag: imageConfig.tag, repo: imageConfig.repo}).$promise;
   };
 
+  service.downloadImages = function(images) {
+    var deferred = $q.defer();
+    var names = ImageHelper.getImagesNamesForDownload(images);
+    Image.download(names).$promise
+    .then(function success(data) {
+      deferred.resolve(data);
+    })
+    .catch(function error(err) {
+      deferred.reject(err);
+    });
+    return deferred.promise;
+  };
+
+  service.uploadImage = function(file) {
```

Comment:
See my comment above regarding the deferred pattern.

If you do not manipulate the response data, the `deferred` pattern is not required here.

You can simply write:

```javascript
service.uploadImage = function(images) {
  return FileUploadService.loadImages(file);
}
```